### ISSUE TITLE
Woocommerce adress

### ISSUE THREAD

**Issue by cenkgulgun** [inspireui/support#4457]
Hello,
1-Web site users adress is not sync with app. How can I show users' addresses in the app?
2-Onboarding is not working. 
3-How can i do default state like default country?
4-I asked before, but I did not get any results. I want to rank categories and products as on the website. How can I do that?

version: 1.7.6+3001

C:\src\flutter\bin\flutter.bat doctor --verbose

[√] Flutter (Channel beta, 1.19.0-4.3.pre, on Microsoft Windows [Version 10.0.18363.959], locale tr-TR)
    • Flutter version 1.19.0-4.3.pre at C:\src\flutter
    • Framework revision 8fe7655ed2 (5 weeks ago), 2020-07-01 14:31:18 -0700
    • Engine revision 9a28c3bcf4
    • Dart version 2.9.0 (build 2.9.0-14.1.beta)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at C:\Users\Cenk\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.0
    • ANDROID_HOME = C:\Users\Cenk\AppData\Local\Android\sdk
    • ANDROID_SDK_ROOT = C:\Users\Cenk\AppData\Local\Android\sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 48.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.47.2)
    • VS Code at C:\Users\Cenk\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.12.2

[√] Connected device (1 available)
    • Redmi Note 8 • 13162dd2 • android-arm64 • Android 9 (API 28)

• No issues found!
Process finished with exit code 0

**_Important Note:_**
- Please help to ready the support policies before creating a new ticket: https://inspireui.com/support-policies/
- Kindly create only *One Ticket* & includes all the issues, that would help us focus to resolve it better. 
- If your all ticket was resolved & closed, but want to create a new ticket, just do simple step by linking to the previous verified ID (you don't need to submit the form again), for example, #Ticket-ID

Thank you so much for your time 😊

**Comment by minhtruong315** [comment]
@cenkgulgun can u send me common folder and admin account to login your website via email: truong@inspireui.com
with subject: https://github.com/inspireui/support/issues/4457

**Comment by cenkgulgun** [comment]
I sent. And finally, when opening the application, the splashscreen opens 3-4 seconds late.
Thanks for everything

**Comment by minhtruong315** [comment]
1-Web site users adress is not sync with app. How can I show users' addresses in the app?
Try to replace fromAuthUser function in models/user/user.dart like this
```
User.fromAuthUser(Map<String, dynamic> json, String _cookie) {
    try {
      cookie = _cookie;
      id = json['id'].toString();
      name = json['displayname'];
      username = json['username'];
      firstName = json['firstname'];
      lastName = json['lastname'];
      email = json['email'];
      picture = json['avatar'];
      nicename = json['nicename'];
      userUrl = json['url'];
      loggedIn = true;
      shipping = Shipping.fromJson(json['shipping']);
      billing = Billing.fromJson(json['billing']);
    } catch (e) {
      print(e.toString());
    }
  }
```

**Comment by cenkgulgun** [comment]
In app the address part is empty. When I click on the Choose address section, it gives the following error. 
![IMG_1348](https://user-images.githubusercontent.com/68585317/89410656-d2007680-d72c-11ea-9b9c-e4ae57085713.jpg)

**Comment by minhtruong315** [comment]
Hi @cenkgulgun please replace models/user/user.dart and try again
[user.dart.zip](https://github.com/inspireui/support/files/5031879/user.dart.zip)

**Comment by minhtruong315** [comment]
Hi @cenkgulgun 2-Onboarding is not working.
Please replace getCurrencyRate in services/woo_commerce.dart like this
```
@override
  Future<Map<String, dynamic>> getCurrencyRate() async {
    try {
      final http.Response response =
          await http.get("$url/wp-json/api/flutter_user/get_currency_rates");
      var body = convert.jsonDecode(response.body);
      if (response.statusCode == 200 && body != null && body is Map) {
        Map<String, dynamic> data = Map.from(body);
        Map<String, dynamic> currency = Map<String, dynamic>();
        data.keys.forEach((key) {
          currency[key] =
              double.parse("${data[key]['rate'] == 0 ? 1 : data[key]['rate']}");
        });
        return currency;
      } else {
        return null;
      }
    } catch (err) {
      return null;
    }
  }
```

**Comment by minhtruong315** [comment]
Hi @cenkgulgun 3-How can i do default state like default country?
I send you updates via the email for this. You can change default state DefaultStateISOCode in common/config/payments.dart

**Comment by minhtruong315** [comment]
@cenkgulgun 4-I asked before, but I did not get any results. I want to rank categories and products as on the website. How can I do that?
Fluxstore doesn't support this.

**Comment by cenkgulgun** [comment]
Thanks for everything. Adress, onboarding and default state is now working very well. I have a few question. 
1- After confirming the order on the payment screen, it switches the order status to pending. The method of payment is cash on delivery. When you select the payment cod , the order status should be processing.

2- Pageview is not working. I did as described but it doesn't work. I uploaded the screenshots.

3- It displays the prices of some products incorrectly on the main screen. It shows properly when it goes to the product page. As far as I understand, it shows the price incorrectly in variable products. I uploaded the screenshots.


4- And finally, when opening the application, the splashscreen opens 2-3 seconds late. I am waiting about 2-3 seconds white screen. After splash screen is coming.

![IMG_1354](https://user-images.githubusercontent.com/68585317/89506743-3d515380-d7d4-11ea-9c47-93ddc45a271e.jpg)
![Screenshot_1](https://user-images.githubusercontent.com/68585317/89506801-4fcb8d00-d7d4-11ea-9cf7-80bfbf2f2c67.png)

![InkedIMG_1355_LI](https://user-images.githubusercontent.com/68585317/89507094-b81a6e80-d7d4-11ea-8362-fded005a9010.jpg)
![InkedIMG_1356_LI](https://user-images.githubusercontent.com/68585317/89507112-be104f80-d7d4-11ea-9f8d-0f4a9b1f3dd1.jpg)
